### PR TITLE
971 get entity param

### PR DESCRIPTION
**Pull request by crbrox**
Implements #971 

@fgalan 
@kzangeli

### CODE REVIEW

File: test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_all_attrs.test [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. POST /v1/updateContext to create 8787GHY entity
+# 02. GET /v2/entities/8787GHY?attrs=speed,brand,plateCountry,madeInCountry,model
```
Comment (fgalan) [inline test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_all_attrs.test]: 02 includes all the attributes of the entity. Let's add a 3rd step to get only a subset, e.g.:

GET /v2/entities/8787GHY?attrs=speed,plateCountry,model

Comment (fgalan) [inline test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_all_attrs.test]: Ah, sorry... I have realized about the other .test file, that seems to be doing exactly that. I will put the comment there :)

NTC

File: test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_one_attr.test [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. POST /v1/updateContext to create 8787GHY entity
+# 02. GET /v2/entities/8787GHY?attrs=speed
```
Comment (fgalan) [inline test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_one_attr.test]: Let's add a test to ask for two attributes (thus testing the "," separator):

GET /v2/entities/8787GHY?attrs=speed,plateCountry,model

File: src/lib/serviceRoutinesV2/getEntity.cpp [hunk 3]
```diff
@@ -57,21 +58,30 @@ std::string getEntity
   ParseData*                 parseDataP
 )
 {
-  std::string  answer;
-  Entity       entity;
+  using namespace std;
```
Comment (kzangeli) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: Not sure I like the using directive inside a function ...

File: src/lib/serviceRoutinesV2/getEntity.cpp [hunk 4]
```diff
@@ -57,21 +58,30 @@ std::string getEntity
...
 
 
-  // 01. Fill in QueryContextRequest
+  // Fill in QueryContextRequest
   parseDataP->qcr.res.fill(compV[2], "", "false", EntityTypeEmptyOrNotEmpty, "");
+  // optional parameter for attributes
+  string attrs = ciP->uriParam["attrs"];
+  if (attrs!="") {
```
Comment (kzangeli) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: opening '{' on its own line

File: src/lib/serviceRoutinesV2/getEntity.cpp [hunk 5]
```diff
@@ -57,21 +58,30 @@ std::string getEntity
...
+    vector<string> attrsV;
+    stringSplit(attrs, ',', attrsV);
+    for(vector<string>::const_iterator it = attrsV.begin(); it != attrsV.end(); ++it) {
```
Comment (kzangeli) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: ditto
And, space before '('   (for is not  function)

File: src/lib/serviceRoutinesV2/getEntity.cpp (hunk 3 above)
Comment (kzangeli) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: NTC  (as discussed in skype)

File: test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_one_attr.test (hunk 2 above)
Comment (crbrox) [inline test/functionalTest/cases/0971_GET_v2_entity_param_attrs/GET_v2_entity_one_attr.test]: Fixed 137796e3f6d706288dd341989364b13810229acb

File: src/lib/serviceRoutinesV2/getEntity.cpp (hunk 5 above)
Comment (crbrox) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: fixed 8611f5b6f987f6c5333330e418c614f54f96e7f3

File: src/lib/serviceRoutinesV2/getEntity.cpp (hunk 4 above)
Comment (crbrox) [inline src/lib/serviceRoutinesV2/getEntity.cpp]: Fixed 321d8d3

**Comment by fgalan** [comment]
LGTM (passing the ball to @kzangeli for final LGTM + merge + delete branch)

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**